plugins/com.redhat.fabric8analytics.lsp.eclipse.ui/src/com/redhat/fabric8analytics/lsp/eclipse/ui/AuthorizeHandler.java

```diff
@@ -47,7 +47,6 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
 			Fabric8AnalysisPreferences.getInstance().setToken(token);
 			ThreeScaleIntegration.getInstance().set3ScalePreferences(token);
 		} catch (UnsupportedEncodingException | RecommenderAPIException | JSONException | StorageException e) {
-			// TODO Auto-generated catch block
```

Comment:
We need proper error handling here. If user clicks "Enable analyses" button and something goes wrong (in this case the 3scale integration) he/she needs to be notified. Also, you need to return from the catch block to end the method otherwise it would set the analyses enabled=true. Something like this:

Fabric8AnalysisPreferences.getInstance().setLSPServerEnabled(false);
MessageDialogUtils.displayErrorMessage("Cannot run analyses because 3scale integration with server has failed");
return null;

Reply:
I agree, I am working on it.